File: 30 Days of Code/Day 19 Interfaces/calculator.cpp
#include <iostream>

using namespace std;

class AdvancedArithmatic {
	public:
		virtual int divisorSum(int n)= 0;
};

// Something wrong in HackerRank's compiler to use std::accumulate()
// Directly add up the divisors.
class Calculator: public AdvancedArithmatic {
	public:
		virtual int divisorSum(int n) override {
			int sum = 0;

			for (int i = 1; i <= n; i++) {
				sum += n%i == 0 ? i : 0;
			}

			return sum;
		}
};

int main()
{
	int num = 0;
	cin >> num;

	AdvancedArithmatic *cal = new Calculator();
	cout << "I implemented: AdvancedArithmatic" << endl;
	cout << cal->divisorSum(num);
	delete cal;
}

File: 30 Days of Code/Day 19 Interfaces/calculator.py
from abc import ABC, abstractmethod

class AdvancedArithmatic(ABC):
    @abstractmethod
    def divisorSum(self, n):
        pass


class Calculator(AdvancedArithmatic):
    def divisorSum(self, n):
        sum = 0
        for i in range(1, n+1):
            if (n % i == 0):
                sum += i
        return sum

n = int(input())
my_calculator = Calculator()
s = my_calculator.divisorSum(n)
print("I implemented: " + type(my_calculator).__bases__[0].__name__)
print(s)